internal/install/recipes/bundler.go

```diff
@@ -50,30 +55,51 @@ func (b *Bundler) CreateAdditionalTargetedBundle(recipes []string) *Bundle {
 	return b.createBundle(recipes, BundleTypes.ADDITIONALTARGETED)
 }
 
-func (b *Bundler) getCoreRecipeNames() []string {
+func (b *Bundler) GetCoreRecipeNames() []string {
 	coreRecipeNames := make([]string, 0, len(coreRecipeMap))
 	for k := range coreRecipeMap {
 		coreRecipeNames = append(coreRecipeNames, k)
 	}
 	return coreRecipeNames
 }
 
+// createBundle creates a new bundle based on the given recipes and bundle type
+// It iterates over the recipes, detects dependencies, and adds recipes to the bundle
 func (b *Bundler) createBundle(recipes []string, bType BundleType) *Bundle {
 	bundle := &Bundle{Type: bType}
 
 	for _, r := range recipes {
 		if d, ok := b.AvailableRecipes.GetRecipeDetection(r); ok {
 			var bundleRecipe *BundleRecipe
+
 			if dualDep, ok := detectDependencies(d.Recipe.Dependencies); ok {
-				dep := updateDependency(dualDep, recipes)
+				dep := b.updateDependency(dualDep, recipes)
 				if dep != nil {
+					log.Debugf("Found dual dependency and selected : %s", dep)
 					d.Recipe.Dependencies = dep
 				} else {
 					log.Debugf("could not process update for dual dependency: %s", dualDep)
 				}
 			}
+
 			bundleRecipe = b.getBundleRecipeWithDependencies(d.Recipe)
+
 			if bundleRecipe != nil {
+				isSuperInput := utils.StringInSlice(types.SuperAgentRecipeName, recipes)
```

Comment:
just a minor comment. I guess naming this as isSuperAgentTargetedInput could be more understandable